float32 x
float32 y
float32 z
float32 p
float32 uncertainty
float32 dx
float32 dy
float32 dz